### PR TITLE
feat: coinbase smart wallet support

### PR DESCRIPTION
**Pull request by 0xmkh**

### CODE REVIEW

File: packages/rainbowkit/src/types/utils.ts [hunk 1]
```diff
@@ -11,3 +13,77 @@ export type Omit<type, keys extends keyof type> = Pick<
...
+      | undefined;
+  }
+>;
+
+export type WindowProvider = {
+  coinbaseWalletExtension?: WalletProvider | undefined;
+  ethereum?: WalletProvider | undefined;
+  phantom?: { ethereum: WalletProvider } | undefined;
+  providers?: any[] | undefined; // Adjust the type as needed
+};
```
Comment (magiziz) [inline packages/rainbowkit/src/types/utils.ts:89]: Here is example from wagmi https://github.com/wevm/wagmi/blob/main/packages/core/src/connectors/injected.ts#L636-L640

Comment (magiziz) [inline packages/rainbowkit/src/types/utils.ts:89]: [Here](https://github.com/wevm/wagmi/blob/main/packages/core/src/connectors/injected.ts#L636-L640) is a example from wagmi

File: packages/rainbowkit/src/window.d.ts
Comment (magiziz) [inline packages/rainbowkit/src/window.d.ts:1]: `wagmi/window` is deprecated

File: packages/rainbowkit/src/connectors/coinbaseWallet.ts
Comment (jxom) [inline packages/rainbowkit/src/connectors/coinbaseWallet.ts:1]: Should be able to use `coinbaseWallet` in Wagmi now.

**Review by DanielSinclair — APPROVED** [review]

**Pull request merged**